### DISCUSSION TITLE
Buttons, AppId and Web browser actions

### DISCUSSION THREAD

**Discussion by Hankdaddy888** [Windos/BurntToast#163] [Q&A]
Some of this might be out of BurntToast scope but here goes. Powershell script appended to these questions.

**Action buttons**: I get these stacked, ill-defined buttons,
![ugly buttons](https://content.screencast.com/users/HenryKumagai/folders/Snagit/media/18892273-6e8f-41f5-8818-eb073ed36f08/04.19.2021-14.43.png)

rather than the nice side-by-side ones
![awesome buttons](https://github.com/Windos/BurntToast/raw/main/Examples/Example03/Example3-Alarm.png)

Can I better control this?

**AppID**: Is there a way to use New-BTAppId to replace  PowerShell AppId when using the New-BTContent > XML > .NET method (so I don't have to install BurntToast on all our Windows 10s)?

**Button action to Web**: Using New-BTButton to link to Website, Web browser pops up to foreground inconsistently. Can I better force Web browser to the foreground when Web-linked button is clicked?

Thank you!

```powershell
Import-Module -name BurntToast

# create text objects for the message content
$text1 = New-BTText -Content 'Yo!'
$text2 = New-BTText -Content "You gots to chill"
    
# reference the company logo that I prestaged in the module folder
$path = (Get-Module -Name BurntToast).ModuleBase
$image1 = New-BTImage -Source 'exclamation.png' -AppLogoOverride -Crop default


# select the audio tone that should be played when the notification displays
$audio1 = New-BTAudio -Path 'C:\Windows\media\Alarm10.wav'


# assemble the notification object
$binding1 = New-BTBinding -Children $text1, $text2 -AppLogoOverride $image1
$visual1 = New-BTVisual -BindingGeneric $binding1
$Button1 = New-BTButton -Content 'Show me how!' -Arguments 'https://showmehow.net' -ActivationType Protocol #A sample URL
$Button2 = New-BTButton -Content 'I''ll do it later' -Dismiss
$Action1 = New-BTAction -Buttons $Button1, $Button2
$content1 = New-BTContent -Visual $visual1 -Audio $audio1 -Actions $Action1 -Scenario IncomingCall

    
$Content1.GetContent()

$CleanContent = $Content1.GetContent().Replace('<text>{', '<text>')
$CleanContent = $CleanContent.Replace('}</text>', '</text>')
$CleanContent = $CleanContent.Replace('="{', '="')
$CleanContent = $CleanContent.Replace('}" ', '" ')

Set-Content -Value $CleanContent -Path 'BT.xml'

$Xml = Get-Content -Path 'BT.xml'

New-BTAppId -AppId 'MyID'

$AppId = 'MyID'
# Commenting PowerShell AppId $AppId = '{1AC14E77-02E7-4E5D-B744-2EB1AE5198B7}\WindowsPowerShell\v1.0\powershell.exe'

$null = [Windows.UI.Notifications.ToastNotificationManager, Windows.UI.Notifications, ContentType = WindowsRuntime]
$null = [Windows.Data.Xml.Dom.XmlDocument, Windows.Data.Xml.Dom.XmlDocument, ContentType = WindowsRuntime]

$ToastXml = [Windows.Data.Xml.Dom.XmlDocument]::new()
$ToastXml.LoadXml($Xml)

$Toast = [Windows.UI.Notifications.ToastNotification]::new($ToastXml)
[Windows.UI.Notifications.ToastNotificationManager]::CreateToastNotifier($AppId).Show($Toast)
```

**Comment by Windos** [comment]
It looks like what you're seeing there with the buttons is a new UX when using the "IncomingCall" scenario (I've never actually come across that and I'm not a fan haha), if you remove `-Scenario IncomingCall` it'll work with the buttons.

If you're using it for the increased duration, try "Alarm" instead.

On the audio front (you didn't ask, but I'll mention it anyway lol), pointing to local files won't work. It broke a few windows versions back. Suggest picking something from the list when you do `New-BTAudio -Source` rather than `-Path`.

For AppIDs, the usefulness of `New-BTAppId` has diminished over the years. Unfortunately, I can't guarantee that using it will work at all anymore.

As for what you CAN use there, run `Get-StartApps` and pick an AppID from the list. Anything there is valid. For example, I could use `{7C5A40EF-A0FB-4BFC-874A-C0F2E0B9FA8E}\Chocolatey GUI\ChocolateyGui.exe` to attribute a toast to Chocolatey GUI, or `agilebits.1password` for 1Password.

![image](https://user-images.githubusercontent.com/6955786/115311729-40566d80-a1c4-11eb-87f9-c9970e7b14bc.png)

**Reply by Windos** [reply]
Oh, and for the focus thing, it's not really something that can be controlled via the toast system itself. But you are using PowerShell, so you can set focus [programmatically](https://stackoverflow.com/questions/42566799/how-to-bring-focus-to-window-by-process-name) (though timing will be your issue here)

**Reply by Hankdaddy888** [reply]
Thank you, Joshua. I incorporated your -Scenario Alarm, New-BTAudio -Source, and Get-StartApps/AppId suggestions.

A couple follow ups:
**AppId at top**: How do you position the AppId above text, as shown in your Chocolatey GUI illustration?

**No AppId text without -AppLogoOverride**: I notice if I don't use -AppLogoOverride, it removes the AppId text (which is preferable). Is there a way to replicate this behavior with -AppLogoOverride?

![No AppId text](https://content.screencast.com/users/HenryKumagai/folders/Snagit/media/75137d0e-0919-4759-a282-93825df6f268/04.20.2021-11.51.png)

**Reply by Windos** [reply]
Correct me if I'm wrong, but you're running an older version of Windows 10 here (i.e. not 20H2)?

If that's the case, you're not seeing the application name at all because you don't have the app logo override on it (the thinking back then was that you're seeing the true app logo, so that was enough to identify the app the toast came from)

![Toast2](https://user-images.githubusercontent.com/6955786/116012169-10eca880-a67d-11eb-99cf-92fd216ba1d4.png)

_App Icon next to text... no "text" attribution to app_

In 20H2 they changed it so that the icon and text identifying the application shows *all* the time app logo override or not. (The positioning of some elements also changed, hence the exact positioning of the icon and text in my example).

![Toast2](https://user-images.githubusercontent.com/6955786/116012185-2feb3a80-a67d-11eb-89a3-d7277d6acf83.png)

_App Icon above text with "text" attribution to app_

![Toast3](https://user-images.githubusercontent.com/6955786/116012264-b56eea80-a67d-11eb-8343-f345ead03477.png)

_This one shows the new layout with app logo and text even with an app logo override_

Those examples were taken pre and post update to 20H2 as I knew there were visual changes and I wanted to be across them lol.

**Reply by Hankdaddy888** [reply]
That makes sense. I am on 2004.

I also read your [latest blog](https://toastit.dev/2021/04/21/road-to-burnttoast-v1-1/). Any ETA on BurntToast v1.0? It also inspired me to get more hands on with the UWP Toast-building. I am hoping this might help actions force apps in to focus and to the foreground.

**Reply by Windos** [reply]
A full release is a fairly long way out, in all honesty (I'm changing jobs which will slow things down too). But I'll be doing preview releases over time so that I can get smaller changes out there even if they aren't complete.

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
